From 46355830a010e30b11d02da939d1b7214dd0e4e0 Mon Sep 17 00:00:00 2001
From: Samin Guo <samin.guo@starfivetech.com>
Date: Fri, 3 Mar 2023 16:59:21 +0800
Subject: [PATCH 54/70] riscv: dts: starfive: jh7110: Add ethernet device nodes

Add JH7110 ethernet device node to support gmac driver for the JH7110
RISC-V SoC.

Signed-off-by: Yanhong Wang <yanhong.wang@starfivetech.com>
Signed-off-by: Samin Guo <samin.guo@starfivetech.com>
---
 arch/riscv/boot/dts/starfive/jh7110.dtsi | 91 ++++++++++++++++++++++++
 1 file changed, 91 insertions(+)

diff --git a/arch/riscv/boot/dts/starfive/jh7110.dtsi b/arch/riscv/boot/dts/starfive/jh7110.dtsi
index 3724c2c9035a..b0e87198808a 100644
--- a/arch/riscv/boot/dts/starfive/jh7110.dtsi
+++ b/arch/riscv/boot/dts/starfive/jh7110.dtsi
@@ -234,6 +234,13 @@ i2srx_lrck_ext: i2srx-lrck-ext-clock {
 		#clock-cells = <0>;
 	};
 
+	stmmac_axi_setup: stmmac-axi-config {
+		snps,lpi_en;
+		snps,wr_osr_lmt = <4>;
+		snps,rd_osr_lmt = <4>;
+		snps,blen = <256 128 64 32 0 0 0>;
+	};
+
 	tdm_ext: tdm-ext-clock {
 		compatible = "fixed-clock";
 		clock-output-names = "tdm_ext";
@@ -658,5 +665,89 @@ mmc1: mmc@16020000 {
 			starfive,sysreg = <&sys_syscon 0x9c 0x1 0x3e>;
 			status = "disabled";
 		};
+
+               gmac0: ethernet@16030000 {
+                       compatible = "starfive,jh7110-dwmac", "snps,dwmac-5.20";
+                       reg = <0x0 0x16030000 0x0 0x10000>;
+                       clocks = <&aoncrg JH7110_AONCLK_GMAC0_AXI>,
+                                <&aoncrg JH7110_AONCLK_GMAC0_AHB>,
+                                <&syscrg JH7110_SYSCLK_GMAC0_PTP>,
+                                <&aoncrg JH7110_AONCLK_GMAC0_TX_INV>,
+                                <&syscrg JH7110_SYSCLK_GMAC0_GTXC>;
+                       clock-names = "stmmaceth", "pclk", "ptp_ref",
+                                     "tx", "gtx";
+                       resets = <&aoncrg JH7110_AONRST_GMAC0_AXI>,
+                                <&aoncrg JH7110_AONRST_GMAC0_AHB>;
+                       reset-names = "stmmaceth", "ahb";
+                       interrupts = <7>, <6>, <5>;
+                       interrupt-names = "macirq", "eth_wake_irq", "eth_lpi";
+                       phy-mode = "rgmii-id";
+                       snps,multicast-filter-bins = <64>;
+                       snps,perfect-filter-entries = <8>;
+                       rx-fifo-depth = <2048>;
+                       tx-fifo-depth = <2048>;
+                       snps,fixed-burst;
+                       snps,no-pbl-x8;
+                       snps,force_thresh_dma_mode;
+                       snps,axi-config = <&stmmac_axi_setup>;
+                       snps,tso;
+                       snps,en-tx-lpi-clockgating;
+                       snps,txpbl = <16>;
+                       snps,rxpbl = <16>;
+                       status = "disabled";
+                       phy-handle = <&phy0>;
+
+                       mdio {
+                               #address-cells = <1>;
+                               #size-cells = <0>;
+                               compatible = "snps,dwmac-mdio";
+
+                               phy0: ethernet-phy@0 {
+                                       reg = <0>;
+                               };
+                       };
+               };
+
+               gmac1: ethernet@16040000 {
+                       compatible = "starfive,jh7110-dwmac", "snps,dwmac-5.20";
+                       reg = <0x0 0x16040000 0x0 0x10000>;
+                       clocks = <&syscrg JH7110_SYSCLK_GMAC1_AXI>,
+                                <&syscrg JH7110_SYSCLK_GMAC1_AHB>,
+                                <&syscrg JH7110_SYSCLK_GMAC1_PTP>,
+                                <&syscrg JH7110_SYSCLK_GMAC1_TX_INV>,
+                                <&syscrg JH7110_SYSCLK_GMAC1_GTXC>;
+                       clock-names = "stmmaceth", "pclk", "ptp_ref",
+                                     "tx", "gtx";
+                       resets = <&syscrg JH7110_SYSRST_GMAC1_AXI>,
+                                <&syscrg JH7110_SYSRST_GMAC1_AHB>;
+                       reset-names = "stmmaceth", "ahb";
+                       interrupts = <78>, <77>, <76>;
+                       interrupt-names = "macirq", "eth_wake_irq", "eth_lpi";
+                       phy-mode = "rgmii-id";
+                       snps,multicast-filter-bins = <64>;
+                       snps,perfect-filter-entries = <8>;
+                       rx-fifo-depth = <2048>;
+                       tx-fifo-depth = <2048>;
+                       snps,fixed-burst;
+                       snps,no-pbl-x8;
+                       snps,force_thresh_dma_mode;
+                       snps,axi-config = <&stmmac_axi_setup>;
+                       snps,tso;
+                       snps,en-tx-lpi-clockgating;
+                       snps,txpbl = <16>;
+                       snps,rxpbl = <16>;
+                       status = "disabled";
+                       phy-handle = <&phy1>;
+
+                       mdio {
+                               #address-cells = <1>;
+                               #size-cells = <0>;
+                               compatible = "snps,dwmac-mdio";
+
+                               phy1: ethernet-phy@1 {
+                                       reg = <0>;
+                               };
+                       };
+               };
 	};
 };
-- 
2.39.2

